src/qibolab/platforms/abstract.py

```diff
@@ -120,8 +125,14 @@ def _check_connected(self):
 
     def reload_settings(self):
         # TODO: Remove ``self.settings``
-        with open(self.runcard) as file:
-            settings = self.settings = yaml.safe_load(file)
+        if len(self.current_config) == 0:
+            # Load initial configuration
+            with open(self.runcard) as file:
+                settings = self.settings = yaml.safe_load(file)
+                self.current_config = self.settings
```

Comment:
What is the motivation of keeping multiple references to the dictionary loaded from the runcard?

Personally I was hoping to clean up the `AbstractPlatform` from all dictionaries and at some stage even remove `self.settings` and replace by proper data structures defined as Python objects, such as the `Qubit` object, the native gate objects in #387 and maybe other objects as needed. 

I believe this is more in line with what @scarrazza initially had in mind when we were refactoring in the past.

Reply:
Thanks @stavros11 for your comments. I am agree that the self.current_config dict is not necessary, and keeping only self.settings should be enough until we replace by data structures some dict. 

We can also recreate the runcard with the class dictionaries attributes when we dump the data into the report. But I think at this point, a dictionary containing all runcard settings is needed if we want to reuse some instrument setup functions, where a dictionary containing all settings is passed. 
For example, after fitting the thresholds or iq_angle in qblox, we need to re-setup the instruments to apply the changes needed for the next step in the autocalibration.